core/xml.js

```diff
@@ -551,6 +576,21 @@ Blockly.Xml.domToBlockHeadless_ = function(xmlBlock, workspace) {
         // Fall through.
       case 'field':
         var field = block.getField(name);
+        if (field instanceof Blockly.FieldVariable) {
+          // TODO (marisaleung): When we change setValue and getValue to
+          // interact with id's instead of names, update this so that we get
+          // the variable based on id instead of textContent.
+          var variable = workspace.getVariable(xmlChild.textContent);
```

Comment:
There is a typeError at  L588.
I've changed it so that the variable is created if it didn't exist before.